### PR TITLE
fix(orb-ui): fix page issues after filter/poll update

### PR DESCRIPTION
**Pull request by gpazuch**
* fixed expand cycle 'dir' param undefined - root cause of issue before deploy
* fixed condition on expand cycle, adopted scan() in favor of reduce()
* fixed keep watching on observable of another page when not needed.
* fixed dataset add component only querying the first 100 items
* fixed group service, group listing page errors
* fixed sink service
* fixed policy service
* fixed dataset service
* fixed agent policy view dataset list
* improved load time for agent policy view

### CODE REVIEW

File: ui/src/app/common/services/orb.service.ts [hunk 1]
```diff
@@ -78,7 +63,20 @@ export class OrbService implements OnDestroy {
   }
 
   observe<T>(observable: Observable<T>) {
-    return this.poller$.pipe(
+    const controller = merge(
```
Comment (gpazuch) [inline ui/src/app/common/services/orb.service.ts:66]: Reusing this controller instance to all observed things was causing issues.
Now, observed things stop being observed after component unsubscribes its 'async' properties.

File: ui/src/app/common/services/orb.service.ts [hunk 2]
```diff
@@ -103,45 +101,27 @@ export class OrbService implements OnDestroy {
...
-    this.sinks$ = this.observe(
-      this.sink.getAllSinks(),
-    );
+  getAgentListView() {
+    return this.observe(this.agent.getAllAgents());
+  }
 
-    this.agentsTags$ = this.agents$.pipe(
+  getAgentsTags() {
+    return this.observe(this.agent.getAllAgents()).pipe(
```
Comment (gpazuch) [inline ui/src/app/common/services/orb.service.ts:124]: once observe() is made into a decorator, I can stop wrapping it around the observed thing and simply add @observe on top of each method.

File: ui/src/app/common/services/orb.service.ts [hunk 3]
```diff
@@ -157,55 +137,80 @@ export class OrbService implements OnDestroy {
...
+    return this.observe(this.group.getAllAgentGroups());
   }
 
   getPolicyListView() {
-    return this.policies$;
+    return this.observe(this.policy.getAllAgentPolicies());
+  }
+
+  getPolicyFullView(id: string) {
+    // retrieve policy
```
Comment (gpazuch) [inline ui/src/app/common/services/orb.service.ts:157]: this is the mapping used to assemble the agent policy view.
it was scattered in agent policy view and its child components.

File: ui/src/app/pages/datasets/add/dataset.add.component.ts [hunk 4]
```diff
@@ -187,11 +218,11 @@ export class DatasetAddComponent {
   getAvailableAgentPolicies() {
     return new Promise((resolve) => {
       this.loading[CONFIG.POLICIES] = true;
-      const pageInfo = { ...AgentPoliciesService.getDefaultPagination(), limit: 100 };
```
Comment (gpazuch) [inline ui/src/app/pages/datasets/add/dataset.add.component.ts:190]: this component was only getting the first 100 policies and sinks for user to choose from when creating/editing dataset - fixed

**Review by brunofariasdeo — APPROVED** [review]
LGTM
I've tested it locally and I believe that all issues mentioned are gone

File: ui/src/app/common/services/orb.service.ts [hunk 5]
```diff
@@ -157,55 +137,80 @@ export class OrbService implements OnDestroy {
...
+    return this.policy.getAgentPolicyById(id).pipe(
+      mergeMap((policy) =>
+        // need a way to get a dataset linked to a policy without having to filter it out
+        this.dataset.getAllDatasets().pipe(
```
Comment (gpazuch) [inline ui/src/app/common/services/orb.service.ts:161]: @lpegoraro would it be feasible to add a filter by 'policy_id' in the dataset list endpoint?

**Comment by gpazuch** [comment]
@brunofariasdeo 

I've pushed fixes to the issue you reported to me about not rendering Agent Policy View page for newly created policies.

While the fixes do solve for this problem, and I've also added extra measure of checking 'agent_group_id' is !== '', it does not fix for one specific case:

* when removing an agent group, it may take a while for dataset to reflect this, causing UI to request a resource that is no longer existent/unretrievable. 

It shouldn't be too hard to add a proper treatment for that, but not without going a bit knee deep on rxjs observables.

I think we are good to merge, and deal with the error in a new 'slab' of code

**Review by brunofariasdeo — APPROVED** [review]
looks great, ty!

**Pull request merged**